### PR TITLE
Fix refresh match page

### PR DESCRIPTION
**Pull request by YinanQiao00**

### CODE REVIEW

File: app/src/main/java/com/example/activeamigo/Algorithm.java [hunk 1]
```diff
@@ -31,14 +32,38 @@ public void fetchUserDocumentsAndProcess(String user1, HashMap<String, String> f
...
                     }
                 });
     }
+    public HashMap<String,String> getCurrentUserData(){
+        return currentUserData;
+    }
+    private static HashMap<String,String> getUser1Data(String user1, List<DocumentSnapshot> userDocuments){
+
+        HashMap<String,String> user1Data = new HashMap<String ,String>();
+//        Log.d()
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/Algorithm.java]: clean this up, consistency

File: app/src/main/java/com/example/activeamigo/Algorithm.java [hunk 2]
```diff
@@ -31,14 +32,38 @@ public void fetchUserDocumentsAndProcess(String user1, HashMap<String, String> f
...
+                        listener.onFetchComplete(matches,currentUserData);
                     } else {
                         Log.e(TAG, "Error getting user documents: ", task.getException());
                         listener.onFetchError(task.getException());
                     }
                 });
     }
+    public HashMap<String,String> getCurrentUserData(){
+        return currentUserData;
+    }
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/Algorithm.java]: add a new line, consistency

File: app/src/main/java/com/example/activeamigo/Algorithm.java [hunk 3]
```diff
@@ -31,14 +32,38 @@ public void fetchUserDocumentsAndProcess(String user1, HashMap<String, String> f
...
+    private static HashMap<String,String> getUser1Data(String user1, List<DocumentSnapshot> userDocuments){
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/Algorithm.java]: add a description of function

File: app/src/main/java/com/example/activeamigo/Algorithm.java [hunk 4]
```diff
@@ -31,14 +32,38 @@ public void fetchUserDocumentsAndProcess(String user1, HashMap<String, String> f
...
+                user1Gender = doc.getString("gender");
+                user1Location = doc.getString("location");
+            }
+        }
+        user1Data.put("exercise",user1Exercise);
+        user1Data.put("location",user1Location);
+        user1Data.put("gender",user1Gender);
+        Log.d("User1Data algo",user1Data.toString());
+        return user1Data;
+    }
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/Algorithm.java:68]: new line line 67

File: app/src/main/java/com/example/activeamigo/Algorithm.java [hunk 5]
```diff
@@ -69,6 +94,7 @@ private static List<HashMap<String,Object>> findSimilarSchedulesForUser(String u
                 break;
             }
         }
+
         if (user1Schedule == null) {
             System.out.println("Warning: No schedule found");
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/Algorithm.java:104]: Log statement?

File: app/src/main/java/com/example/activeamigo/ui/matches/MatchesPage.java [hunk 6]
```diff
@@ -22,34 +26,41 @@
...
     private FragmentMatchesBinding binding;
     private ListView matchesList;
     HashMap<String, String[]> matchedUsers = new HashMap<String, String[]>();
+    Algorithm algorithm;
+    MatchesViewModel matchesViewModel;
+    HashMap<String, String> Filters;
+    Spinner exerciseSpinner;
+    Spinner locationSpinner;
+    Spinner genderSpinner;
+    Spinner daySpinner;
```
Comment (cknguyen123) [inline app/src/main/java/com/example/activeamigo/ui/matches/MatchesPage.java]: define access modifier, private or protected

**Review by cknguyen123 — COMMENTED** [review]
Functionality is good, try to add comment and keep styling consistent

File: app/src/main/java/com/example/activeamigo/Algorithm.java (hunk 2 above)
Comment (YinanQiao00) [reply app/src/main/java/com/example/activeamigo/Algorithm.java]: not used function, removed

File: app/src/main/java/com/example/activeamigo/ui/matches/MatchesPage.java (hunk 6 above)
Comment (YinanQiao00) [reply app/src/main/java/com/example/activeamigo/ui/matches/MatchesPage.java]: fixed

**Comment by amquach00** [comment]
lgtm

**Pull request merged**